commit f1671f9a3150d98847016e09fbb983928362c2b1
Author: pchunt <peter.hunt@opengear.com>
Date:   Tue Jun 27 16:17:33 2017 -0600

    OG-4237 ModemManager: support 3GPP2 APNs

    Allow an APN to be specified for 3GPP2 connections.

    Even though its not needed for cdma-evdo, some US carriers need
    to be able to fall back from LTE to those types and will have an
    APN specified.

commit dd535e7af3d0fcf9daff2e0874ea3c0c89a8c643
Author: pchunt <peter.hunt@opengear.com>
Date:   Thu Oct 5 16:51:52 2017 -0600

    OG-4741 Be sure to dereference modem handle if 3GPP2 and APN is set.

Index: ModemManager/src/mm-bearer-qmi.c
===================================================================
--- ModemManager.orig/src/mm-bearer-qmi.c
+++ ModemManager/src/mm-bearer-qmi.c
@@ -1821,19 +1821,6 @@ _connect (MMBaseBearer *_self,
         goto out;
     }
 
-    /* Is this a 3GPP2 only modem and APN was given? If so, error */
-    if (mm_iface_modem_is_cdma_only (MM_IFACE_MODEM (modem)) && apn) {
-        g_task_report_new_error (
-            self,
-            callback,
-            user_data,
-            _connect,
-            MM_CORE_ERROR,
-            MM_CORE_ERROR_INVALID_ARGS,
-            "3GPP2 doesn't support APN setting");
-        goto out;
-    }
-
     mm_obj_dbg (self, "launching connection with QMI port (%s) and data port (%s)",
                 mm_port_get_device (MM_PORT (qmi)),
                 mm_port_get_device (data));
Index: ModemManager/src/mm-broadband-bearer.c
===================================================================
--- ModemManager.orig/src/mm-broadband-bearer.c
+++ ModemManager/src/mm-broadband-bearer.c
@@ -1299,20 +1299,6 @@ connect (MMBaseBearer *self,
         return;
     }
 
-    /* Is this a 3GPP2 only modem and APN was given? If so, error */
-    if (mm_iface_modem_is_cdma_only (MM_IFACE_MODEM (modem)) && apn) {
-        g_task_report_new_error (
-            self,
-            callback,
-            user_data,
-            connect,
-            MM_CORE_ERROR,
-            MM_CORE_ERROR_INVALID_ARGS,
-            "3GPP2 doesn't support APN setting");
-        g_object_unref (modem);
-        return;
-    }
-
     /* In this context, we only keep the stuff we'll need later */
     task = g_task_new (self, cancellable, callback, user_data);
 
@@ -1332,7 +1318,7 @@ connect (MMBaseBearer *self,
     }
 
     /* Otherwise, launch CDMA-specific connection. */
-    if (mm_iface_modem_is_cdma (MM_IFACE_MODEM (modem)) && !apn) {
+    if (mm_iface_modem_is_cdma (MM_IFACE_MODEM (modem))) {
         mm_obj_dbg (self, "launching 3GPP2 connection attempt");
         MM_BROADBAND_BEARER_GET_CLASS (self)->connect_cdma (
             MM_BROADBAND_BEARER (self),
